### PR TITLE
Update to 3.1 package baseline

### PR DESCRIPTION
**Pull request by pakrym**
What can go wrong?

### CODE REVIEW

File: sdk/core/Azure.Core/src/Azure.Core.csproj [hunk 1]
```diff
@@ -33,6 +33,11 @@
     <PackageReference Include="System.Memory.Data" />
   </ItemGroup>
 
+  <ItemGroup Condition="'$(TargetFramework)' == 'net461'">
```
Comment (pakrym) [inline sdk/core/Azure.Core/src/Azure.Core.csproj:36]: Why do I suddenly need this?

Comment (weshaggard) [reply sdk/core/Azure.Core/src/Azure.Core.csproj:36]: Maybe they were transitive dependencies from some of the other packages before but aren't any longer.

**Review by weshaggard — APPROVED** [review]
Looks reasonable but we should probably verify that the runtime dependency changes don't impact PS or Azure functions.

**Review by jsquire — APPROVED** [review]

**Comment by pakrym** [comment]
Compared with https://pscoretestdata.blob.core.windows.net/ps-deps-json/servicing/pwsh.deps.json. We are good there. Looking into Functions.

**Comment by pakrym** [comment]
compared to "c:\Program Files\Microsoft\Azure Functions Core Tools\func.deps.json" we are good there as well.

**Comment by pakrym** [comment]
@heaths, can you, please, help me sanity check these version changes?

File: eng/Packages.Data.props [hunk 2]
```diff
@@ -211,14 +213,14 @@
     <PackageReference Update="Microsoft.Azure.Test.HttpRecorder" Version="[1.13.3, 2.0.0)" />
     <PackageReference Update="Microsoft.Azure.WebJobs.Extensions" Version="4.0.1" />
     <PackageReference Update="Microsoft.Azure.WebJobs.Extensions.Http" Version="3.0.2" />
-    <PackageReference Update="Microsoft.CSharp" Version="4.6" />
+    <PackageReference Update="Microsoft.CSharp" Version="4.7" />
```
Comment (heaths) [inline eng/Packages.Data.props:216]: Nit: might be equivalent, but should go with the full semver to avoid ambiguity i.e. 4.7.0.

**Review by heaths — APPROVED** [review]
The updated/added versions are the latest within the range of supported TFMs, so assuming there's no CVEs active against them (nothing I found in Component Governance) LGTM.

File: eng/Packages.Data.props (hunk 2 above)
Comment (pakrym) [inline eng/Packages.Data.props:216]:
```suggestion
    <PackageReference Update="Microsoft.CSharp" Version="4.7.0" />
```

**Review by heaths — APPROVED** [review]

**Pull request merged**